python/cudf/cudf/core/column/column.py

```diff
@@ -800,10 +801,10 @@ def unique_count(self, method="sort", dropna=True):
         if method != "sort":
             msg = "non sort based unique_count() not implemented yet"
             raise NotImplementedError(msg)
-        segs, _ = self._unique_segments()
-        if dropna is False and self.null_count > 0:
-            return len(segs) + 1
-        return len(segs)
+        count = cpp_unique_count(self)
+        if dropna is True and self.null_count > 0:
+            return count - 1
```

Comment:
Ideally these two lines could be somehow moved into the C++ layer and parameterized for handling `null`?

Reply:
@karthikeyann do you plan to address this?

Reply:
addressed now.